src/main/java/com/onarandombox/multiverseinventories/MultiverseInventories.java

```diff
@@ -220,6 +224,22 @@ private void hookImportables() {
         }
     }
 
+    private void hookLuckPerms() {
+        if (Bukkit.getPluginManager().getPlugin("LuckPerms") == null) {
+            Logging.finer("LuckPerms is not installed on your server.");
+            return;
+        }
+        LuckPerms luckPerms;
+        try {
+            luckPerms = LuckPermsProvider.get();
+        } catch (IllegalArgumentException e) {
+            Logging.warning("Unable to hook into LuckPerms. API not enabled!");
+            Logging.warning(e.getMessage());
```

Comment:
Is the API something that has to be enabled by the user? Or is this an actual error?

Just wondering because it might be annoying if someone purposefully disables the API and gets a warning every time they start their server.

Reply:
If luckperms is installed (checking alr done above), it should always be enabled unless luckperms crashed on their server.

Reply:
Okay, looks good then.

Reply:
Actually I just took a look at ContextCalculator's Javadoc, do you know if the calculation is thread-safe?

Reply:
it needs to be thread-safe bcu calls may be async. Source code has 3 pointers to take note:

https://github.com/lucko/LuckPerms/blob/5ea2aee2f49a317e8b6aa242ded2fe81384343ed/api/src/main/java/net/luckperms/api/context/ContextCalculator.java#L39-L53

Reply:
Yeah that's what I read, I was asking if the code you wrote is thread-safe.

Reply:
https://github.com/Multiverse/Multiverse-Inventories/blob/f704fa2fef6fc2a2693e98271de159c6e7274636/src/main/java/com/onarandombox/multiverseinventories/AbstractWorldGroupManager.java#L54-L60

hmm yea `getGroupsForWorld` does have the chance of modifying... I think using `getGroups` is a safer option since makes a copy